gnuradio-runtime/lib/buffer.cc

```diff
@@ -109,25 +106,27 @@ namespace gr {
    * returns true iff successful.
    */
   bool
-  buffer::allocate_buffer(int nitems, size_t sizeof_item)
+  buffer::allocate_buffer()
   {
-    int orig_nitems = nitems;
+    unsigned int nitems = d_bufsize;
+    int orig_nitems = d_bufsize;
 
     // Any buffersize we come up with must be a multiple of min_nitems.
     int granularity = gr::vmcircbuf_sysconfig::granularity();
-    int min_nitems =  minimum_buffer_items(sizeof_item, granularity);
+    int min_nitems =  minimum_buffer_items(d_sizeof_item, granularity);
 
     // Round-up nitems to a multiple of min_nitems.
-    if(nitems % min_nitems != 0)
+    if(nitems % min_nitems != 0) {
       nitems = ((nitems / min_nitems) + 1) * min_nitems;
+    }
 
     // If we rounded-up a whole bunch, give the user a heads up.
     // This only happens if sizeof_item is not a power of two.
 
-    if(nitems > 2 * orig_nitems && nitems * (int) sizeof_item > granularity){
+    if(nitems > 2 * orig_nitems && nitems * static_cast<int>(d_sizeof_item) > granularity){
       std::cerr << "gr::buffer::allocate_buffer: warning: tried to allocate\n"
```

Comment:
This would be a good time to make the change to using logging instead of std::cerr.  I wouldn't hold this PR up for that, but it's something that needs doing eventually.